### ISSUE TITLE
Add distributed support for ROC_AUC metric

### ISSUE THREAD

**Issue by Nic-Ma** [Project-MONAI/MONAI#869]
**Is your feature request related to a problem? Please describe.**
As ignite doesn't support distributed training for `epoch-gather` metrics, we need to enhance `ROCAUC` referring to:
https://pytorch.org/docs/stable/distributed.html#torch.distributed.gather

**Comment by wyli** [comment]
this ticket needs the feedback/confirmation from the working group
